transform/src/test/scala/amf/transform/canonical/CanonicalWebAPISpecDialectTest.scala

```diff
@@ -57,27 +80,21 @@ class CanonicalWebAPISpecDialectTest extends FunSuiteCycleTests with PlatformSec
     }
   }
 
-  val tests: Seq[String] = Seq(
-    "simple/api.raml",
-    "annotations/api.raml",
-    "macros/api.raml",
-    "modular/api.raml",
-    "security/api.raml",
-    "declares/api.raml",
-    "tuple-shape-schema/api.raml"
-    //    "modular-recursion/api.raml"
-  )
-
-  tests.foreach { input =>
-    val golden = input.replace("api.raml", "webapi")
-    test(s"Test '$input' for WebAPI dialect transformation and yaml/json rendering") {
-      checkCanonicalDialectTransformation(input, golden, shouldTranform = false)
+  private def canonicalTransform(webApiPath: String, hint: Hint = RamlYamlHint) =
+    for {
+      _           <- AMF.init()
+      _           <- Validation(platform)
+      _           <- registerCanonicalDialect()
+      unit        <- AMFCompiler(webApiPath, platform, hint, eh = DefaultParserErrorHandler()).build()
+      transformed <- CanonicalWebAPISpecTransformer.transform(unit)
+    } yield {
+      transformed
     }
-  }
 
-  test("Test that canonical transformer only accepts Documents") {
-    val unit = ExternalFragment()
-    assertThrows[DocumentExpectedException](CanonicalWebAPISpecTransformer.transform(unit))
-  }
+  private def registerCanonicalDialect(): Future[Unit] =
+    for {
+      _ <- Future(amf.Core.registerPlugin(AMLPlugin))
+      d <- AMFCompiler(CANONICAL_WEBAPI_DIALECT, platform, VocabularyYamlHint, eh = UnhandledParserErrorHandler).build()
+      _ <- Future { AMLPlugin().registry.resolveRegisteredDialect(d.asInstanceOf[Dialect].header) }
+    } yield {}
```

Comment:
You can simply yield `AMLPlugin().registry.resolveRegisteredDialect(d.asInstanceOf[Dialect].header)`